homeassistant/components/influxdb.py

```diff
@@ -220,9 +221,11 @@ def event_to_json(event):
                         json['fields'][key] = float(
                             RE_DECIMAL.sub('', new_value))
 
-                # Infinity is not a valid float in InfluxDB
-                if (key, float("inf")) in json['fields'].items():
```

Comment:
Whoa, this line was pretty inefficient. If I understand correctly this would previously have stepped through the entire dictionary. It's not a big data structure but still...

Reply:
I thought it would be efficient because `items()` creates a view object, not a list.

Reply:
It still has to iterate through all items. New check is constant time 👍

Reply:
Hmm, I looked it up now and I still think the previous implementation was just as constant.

PEP 3106 says that (Python 3) `.items()` is "set-like" and specifies `__contains__` with this pseudocode:
```python
    def __contains__(self, (key, value)):
        return key in self.__d and self.__d[key] == value
```

Reply:
Oh wow, thanks for the info. You learn something new every day :D